### DISCUSSION TITLE
Using slots in TGSL

### DISCUSSION THREAD

**Discussion by iwoplaza** [software-mansion/TypeGPU#586] [Ideas]
Lets take the following code as an example:
```ts
const inputGridSlot = tgpu.slot<TgpuBufferReadonly<TgpuArray<F32>>>();

const getCell = tgpu.fn([i32, i32], f32).does((x, y) => {
  const idx = x + y * gridSize;
  const gridUsage = inputGridSlot.value; // => TgpuBufferReadonly<TgpuArray<F32>>
  const grid = gridUsage.value; // => number[];
  return grid[idx]; // => number
});
```

To get the actual resource, we need to perform two `.value` accesses.

I see two paths we could take:
- Leave it, its fine
- Make `.value` act as `.value.value` when done on slots
- (something else)

**Comment by reczkok** [comment]
I think having `.value` act as `.value.value` would be the most convenient. We could also introduce a field like `.current` in case someone wants to use the current slot occupant in some logic outside of shader functions.
